File: src/main/kotlin/com/coin/upbit/upbit/domain/BaseDomain.java
package com.coin.upbit.upbit.domain;

import com.sun.istack.NotNull;
import kotlin.jvm.internal.Intrinsics;

import javax.persistence.Column;
import javax.persistence.MappedSuperclass;
import javax.persistence.PrePersist;
import javax.persistence.PreUpdate;
import java.time.LocalDateTime;

@MappedSuperclass
public abstract class BaseDomain {
   @Column(
      name = "CREATED_AT",
      updatable = false
   )
   @NotNull
   public LocalDateTime createdAt;
   @Column(
      name = "UPDATED_AT")
   @NotNull
   public LocalDateTime updatedAt;

   @PrePersist
   public final void prePersist() {
      LocalDateTime var10001 = LocalDateTime.now();
      Intrinsics.checkNotNullExpressionValue(var10001, "LocalDateTime.now()");
      this.createdAt = var10001;
      var10001 = LocalDateTime.now();
      Intrinsics.checkNotNullExpressionValue(var10001, "LocalDateTime.now()");
      this.updatedAt = var10001;
   }

   @PreUpdate
   public final void preUpdate() {
      LocalDateTime var10001 = LocalDateTime.now();
      Intrinsics.checkNotNullExpressionValue(var10001, "LocalDateTime.now()");
      this.updatedAt = var10001;
   }
}

File: src/main/kotlin/com/coin/upbit/global/domain/BaseDomain.kt
//package com.coin.upbit.global.domain
//
//import java.time.LocalDateTime
//import javax.persistence.Column
//import javax.persistence.MappedSuperclass
//import javax.persistence.PrePersist
//import javax.persistence.PreUpdate
//
//@MappedSuperclass
//abstract class BaseDomain {
//    @Column(name = "CREATED_AT", columnDefinition = "DATE", updatable = false)
//    lateinit var createdAt: LocalDateTime
//
//    @Column(name = "UPDATED_AT", columnDefinition = "DATE")
//    lateinit var updatedAt: LocalDateTime
//
//    @PrePersist
//    fun prePersist() {
//        this.createdAt = LocalDateTime.now()
//        this.updatedAt = LocalDateTime.now()
//    }
//
//    @PreUpdate
//    fun preUpdate() {
//        this.updatedAt = LocalDateTime.now()
//    }
//}